### PR TITLE
fix: added job credentials on PipelineJobSchedule init

### LINKED ISSUE #4039: Missing credentials on schedule creation based on PipelineJob
When creating a PipelineJob object I would assume that project, location and credentials are also used for the Schedule that i can create of this job. However when I use the `create_schedule()` method this is not the case. The stacktrace below shows the behavior for missing credentials, but the initial reason I started looking into this were reports, that schedules were happening in different locations than ad-hoc triggered vertex jobs.

#### Environment details

  - OS type and version: Ubuntu 24.04
  - Python version: 3.11.9
  - pip version: 24.0
  - `google-cloud-aiplatform` version: 1.56.0

#### Steps to reproduce

  1. Configure a PipelineJob
  2. Create a schedulel from this PipelineJob

#### Code example

```python
import aiplatform

CREDENTIALS = "..."
PROJECT = "..."
LOCATION = "..."

run = aiplatform.PipelineJob(
    display_name="Example",
    template_path="...",
    job_id="...",
    pipeline_root="...",
    parameter_values=None,
    enable_caching=False,
    encryption_spec_key_name="...",
    credentials=CREDENTIALS,
    project=PROJECT,
    location=LOCATION,
)

run.create_schedule(
    display_name="Example_Schedule",
    cron=""*/5 * * * *"",
    start_time="...",
    end_time="...",
    service_account="...",
    network="...",
)
```

#### Stack trace
I apologize for the formatting of my logs, this seems to break down when copy pasting.

```
│ ❱ 606 │   │   │   │   run.create_schedule(                                                       │
│   607 │   │   │   │   │   display_name=schedule.name,                                            │
│   608 │   │   │   │   │   cron=schedule.cron_expression,                                         │
│   609 │   │   │   │   │   start_time=schedule.utc_start_time,                                    │
│                                                                                                  │
│ /home/alexej/PycharmProjects/rivian-demo/venv/lib/python3.11/site-packages/google/cloud/aiplatfo │
│ rm/pipeline_jobs.py:558 in create_schedule                                                       │
│                                                                                                  │
│    555 │   │   │   display_name = self._generate_display_name(prefix="PipelineJobSchedule")      │
│    556 │   │   utils.validate_display_name(display_name)                                         │
│    557 │   │                                                                                     │
│ ❱  558 │   │   pipeline_job_schedule = aiplatform.PipelineJobSchedule(                           │
│    559 │   │   │   pipeline_job=self,                                                            │
│    560 │   │   │   display_name=display_name,                                                    │
│    561 │   │   )                                                                                 │
│                                                                                                  │
│ /home/alexej/PycharmProjects/rivian-demo/venv/lib/python3.11/site-packages/google/cloud/aiplatfo │
│ rm/pipeline_job_schedules.py:88 in __init__                                                      │
│                                                                                                  │
│    85 │   │                                                                                      │
│    86 │   │   project = project or pipeline_job.project                                          │
│    87 │   │   location = location or pipeline_job.location                                       │
│ ❱  88 │   │   super().__init__(credentials=credentials, project=project, location=location)      │
│    89 │   │                                                                                      │
│    90 │   │   self._parent = initializer.global_config.common_location_path(                     │
│    91 │   │   │   project=project, location=location                                             │
│                                                                                                  │
│ /home/alexej/PycharmProjects/rivian-demo/venv/lib/python3.11/site-packages/google/cloud/aiplatfo │
│ rm/schedules.py:74 in __init__                                                                   │
│                                                                                                  │
│    71 │   │   │   │   Optional. Location to create Schedule. If not set,                         │
│    72 │   │   │   │   location set in aiplatform.init will be used.                              │
│    73 │   │   """                                                                                │
│ ❱  74 │   │   super().__init__(project=project, location=location, credentials=credentials)      │
│    75 │                                                                                          │
│    76 │   @classmethod                                                                           │
│    77 │   def get(                                                                               │
│                                                                                                  │
│ /home/alexej/PycharmProjects/rivian-demo/venv/lib/python3.11/site-packages/google/cloud/aiplatfo │
│ rm/base.py:1234 in __init__                                                                      │
│                                                                                                  │
│   1231 │   │   │   │   resource noun.                                                            │
│   1232 │   │   │   resource_name(str): A fully-qualified resource name or ID.                    │
│   1233 │   │   """                                                                               │
│ ❱ 1234 │   │   _VertexAiResourceNounPlus.__init__(                                               │
│   1235 │   │   │   self,                                                                         │
│   1236 │   │   │   project=project,                                                              │
│   1237 │   │   │   location=location,                                                            │
│                                                                                                  │
│ /home/alexej/PycharmProjects/rivian-demo/venv/lib/python3.11/site-packages/google/cloud/aiplatfo │
│ rm/base.py:558 in __init__                                                                       │
│                                                                                                  │
│    555 │   │                                                                                     │
│    556 │   │   self.project = project or initializer.global_config.project                       │
│    557 │   │   self.location = location or initializer.global_config.location                    │
│ ❱  558 │   │   self.credentials = credentials or initializer.global_config.credentials           │
│    559 │   │                                                                                     │
│    560 │   │   appended_user_agent = None                                                        │
│    561 │   │   if base_constants.USER_AGENT_SDK_COMMAND:                                         │
│                                                                                                  │
│ /home/alexej/PycharmProjects/rivian-demo/venv/lib/python3.11/site-packages/google/cloud/aiplatfo │
│ rm/initializer.py:348 in credentials                                                             │
│                                                                                                  │
│   345 │   │   logger = logging.getLogger("google.auth._default")                                 │
│   346 │   │   logging_warning_filter = utils.LoggingFilter(logging.WARNING)                      │
│   347 │   │   logger.addFilter(logging_warning_filter)                                           │
│ ❱ 348 │   │   self._set_project_as_env_var_or_google_auth_default()                              │
│   349 │   │   credentials = self._credentials                                                    │
│   350 │   │   logger.removeFilter(logging_warning_filter)                                        │
│   351 │   │   return credentials                                                                 │
│                                                                                                  │
│ /home/alexej/PycharmProjects/rivian-demo/venv/lib/python3.11/site-packages/google/cloud/aiplatfo │
│ rm/initializer.py:94 in _set_project_as_env_var_or_google_auth_default                           │
│                                                                                                  │
│    91 │   │   │   │   │   )                                                                      │
│    92 │   │   │   │   │   self._project = project_number                                         │
│    93 │   │   │   else:                                                                          │
│ ❱  94 │   │   │   │   credentials, project = google.auth.default()                               │
│    95 │   │   │   │   self._credentials = self._credentials or credentials                       │
│    96 │   │   │   │   self._project = project                                                    │
│    97                                                                                            │
│                                                                                                  │
│ /home/alexej/PycharmProjects/rivian-demo/venv/lib/python3.11/site-packages/google/auth/_default. │
│ py:691 in default                                                                                │
│                                                                                                  │
│   688 │   │   │   │   )                                                                          │
│   689 │   │   │   return credentials, effective_project_id                                       │
│   690 │                                                                                          │
│ ❱ 691 │   raise exceptions.DefaultCredentialsError(_CLOUD_SDK_MISSING_CREDENTIALS)               │
│   692                                                                                            │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
DefaultCredentialsError: Your default credentials were not found. To set up Application Default Credentials, see 
https://cloud.google.com/docs/authentication/external/set-up-adc for more information.

```

### PR DESCRIPTION
**Pull request by AlexejPenner**
This addresses the following [Issue](https://github.com/googleapis/python-aiplatform/issues/4039). When creating a PipelineJob object, I would expect and Schedules created from this job to inherit credentials, location and project from the job.

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-aiplatform/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)
- [ ] Get the necessary approvals
- [ ] Once the last commit on the PR has been approved, add the "ready to pull" label to the Pull Request

Note: do not merge your PR from GitHub. Adding the "ready to pull" label is the final step in the review process.
After approvals, the changes in your PR will be committed to the `main` branch and this PR will be closed.

Fixes #4039 🦕

### CODE REVIEW

**Review by eliottiti — APPROVED** [review]

**Review by jaycee-li — APPROVED** [review]
Thanks @AlexejPenner for the fix!

File: google/cloud/aiplatform/pipeline_jobs.py [hunk 1]
```diff
@@ -558,6 +558,9 @@ def create_schedule(
         pipeline_job_schedule = aiplatform.PipelineJobSchedule(
             pipeline_job=self,
             display_name=display_name,
+            credentials=self.credentials,
+            location=self.location,
+            project=self.project
```
Comment (jaycee-li) [inline google/cloud/aiplatform/pipeline_jobs.py:563]: `project=self.project,`

**Review by jaycee-li — CHANGES_REQUESTED** [review]

File: google/cloud/aiplatform/pipeline_jobs.py (hunk 1 above)
Comment (strickvl) [inline google/cloud/aiplatform/pipeline_jobs.py:563]:
```suggestion
            project=self.project,
```

**Review by jaycee-li — APPROVED** [review]

**Comment by jaycee-li** [comment]
Hi @strickvl , could you please sign the CLA so that we can merge this PR.

https://github.com/googleapis/python-aiplatform/pull/4040/checks?check_run_id=27174525571

**Pull request closed without merging**